File: 1768.merge-strings-alternately/solution.kt
import kotlin.math.*
class Solution {
    fun mergeAlternately(word1: String, word2: String): String {
        var i = 0
        val len1 = word1.length
        val len2 = word2.length
        var limit = min(len1, len2)

        var result = ""
        while (i < limit) {
            result += "${word1[i]}${word2[i]}"
            i++
        }

        limit = max(len1, len2)
        when {
            i < len1 -> return result + word1.substring(i)
            i < len2 -> return result + word2.substring(i)
            else -> return result
        }

    }
}

File: 1768.merge-strings-alternately/solution.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        result = ""
        i = 0
        len1, len2 = len(word1), len(word2)
        while (i < len1) and (i < len2):
            result += word1[i] + word2[i]
            i += 1
        if (i < len1):
            result += word1[i:]
        elif (i < len2):
            result += word2[i:]
        return result
